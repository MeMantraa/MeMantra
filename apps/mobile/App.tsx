import React, { useEffect } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import MainNavigator from './app/index';
import './global.css';

declare const __DEV__: boolean;

export default function App() {
  // Log the current mode
  console.log('In development mode:', __DEV__);

  useEffect(() => {
    // Only initialize Sentry in production/release mode

    if (__DEV__) {
      console.log('Development mode: Sentry disabled');
    } else {
      console.log('Production mode: Initializing Sentry...');
      import('./sentry').then(({ startPerformanceTracking }) => {
        // Start manual performance tracking
        startPerformanceTracking();
      });
    }
  }, []);

  return (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <NavigationContainer>
        <MainNavigator />
      </NavigationContainer>
    </GestureHandlerRootView>
  );
}
