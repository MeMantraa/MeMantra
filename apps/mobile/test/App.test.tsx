import React from 'react';
import { render } from '@testing-library/react-native';
import * as renderer from 'react-test-renderer';
import App from '../App'; // Adjust import path

// Mock any native dependencies
jest.mock('expo-status-bar', () => ({
  StatusBar: 'StatusBar',
}));

describe('App Component', () => {
  test('renders without crashing', () => {
    expect(() => render(<App />)).not.toThrow();
  });

  test('renders the main component', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toBeTruthy();
  });

  test('creates a snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
