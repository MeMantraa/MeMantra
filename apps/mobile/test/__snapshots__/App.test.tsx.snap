// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component creates a snapshot 1`] = `null`;
